File: java/src/com/es0/Produttore.java
package com.es0;

import java.io.BufferedReader;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;

public class Produttore {

	//Produttore <fileDaScrivere>
	
	public static void main(String[] args) {
		if(args.length != 1) {
			System.out.println("Usage: Produttore <FileDaScrivere>");
			System.exit(0);
		}
		
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		FileWriter writer;
		int ch;
		
		try {
			writer = new FileWriter(args[0]);
			
			while((ch = in.read()) != -1) //se leggo -1 ho raggiunto EOF
				writer.write(ch);
			
			writer.close();
			
		} catch (IOException e) {
			e.printStackTrace();
			System.exit(1);
		}
		
	}

}

File: c/Produttore.c
#include <stdio.h>
#include <fcntl.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

//produttore filedascrivere.txt

#define INV_ERR 1
#define OPEN_ERR 2
#define WR_ERR 3

int main(int argc, char *argv[]){

    if(argc != 2){
        perror("Invocazione: produttore filedascrivere.txt");
        exit(INV_ERR);
    }

    int fd = open(argv[1], O_WRONLY | O_CREAT | O_TRUNC, 0640);
    if(fd < 0){
        perror("P0: Impossibile creare/aprire il file");
        exit(OPEN_ERR);
    }

    //leggo fino ad EOF
    char ch;
    while((read(0, &ch, sizeof(char))) > 0){
        if(write(fd, &ch, sizeof(char)) < 0){
            perror("P0: Errore scrittura nel file");
            exit(WR_ERR);
        }
    }

    close(fd);
    exit(0);
}